tests/cache/scalar.rs

```diff
@@ -208,18 +214,1121 @@ mod increment {
     }
 }
 
-mod set_if_not_exists {}
+mod set_if_absent {
+    use super::*;
+
+    #[tokio::test]
+    async fn invalid_cache_name() -> MomentoResult<()> {
+        let client = &CACHE_TEST_STATE.client;
+        let result = client
+            .set_if_absent("   ", "key", "value")
+            .await
+            .unwrap_err();
+        assert_eq!(result.error_code, MomentoErrorCode::InvalidArgumentError);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn nonexistent_cache() -> MomentoResult<()> {
+        let client = &CACHE_TEST_STATE.client;
+        let cache_name = unique_string("fake-cache");
+        let result = client
+            .set_if_absent(cache_name, "key", "value")
+            .await
+            .unwrap_err();
+        assert_eq!(result.error_code, MomentoErrorCode::NotFoundError);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn string_key_string_value() -> MomentoResult<()> {
+        let client = &CACHE_TEST_STATE.client;
+        let cache_name = CACHE_TEST_STATE.cache_name.as_str();
+        let item1 = TestScalar::new();
+        let item2 = TestScalar::new();
+
+        // Setting a key that doesn't exist should create it
+        let result = client
+            .set_if_absent(cache_name, item1.key(), item1.value())
+            .await?;
+        assert_eq!(result, SetIfAbsent::Stored);
+        let result_value: String = client.get(cache_name, item1.key()).await?.try_into()?;
+        assert_eq!(result_value, item1.value());
+
+        // Setting a key that exists should not overwrite it
+        let result = client
+            .set_if_absent(cache_name, item1.key(), item2.value())
+            .await?;
+        assert_eq!(result, SetIfAbsent::NotStored);
+        let result_value: String = client.get(cache_name, item1.key()).await?.try_into()?;
+        assert_eq!(result_value, item1.value());
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn byte_key_byte_value() -> MomentoResult<()> {
+        let client = &CACHE_TEST_STATE.client;
+        let cache_name = CACHE_TEST_STATE.cache_name.as_str();
+        let item1 = TestScalar::new();
+        let item2 = TestScalar::new();
+
+        // Setting a key that doesn't exist should create it
+        let result = client
+            .set_if_absent(cache_name, item1.key(), item1.value())
+            .await?;
+        assert_eq!(result, SetIfAbsent::Stored);
+        let result_value: Vec<u8> = client.get(cache_name, item1.key()).await?.try_into()?;
+        assert_eq!(result_value, item1.value().as_bytes());
+
+        // Setting a key that exists should not overwrite it
+        let result = client
+            .set_if_absent(cache_name, item1.key(), item2.value())
+            .await?;
+        assert_eq!(result, SetIfAbsent::NotStored);
+        let result_value: Vec<u8> = client.get(cache_name, item1.key()).await?.try_into()?;
+        assert_eq!(result_value, item1.value().as_bytes());
```

Comment:
I understand the intent here. In other repos where we exhaustively tested input types, we tested passing a bytes key and string value, bytes key and bytes value, etc.

To contrast here the data in `TestScalar` is always `String` based, so we are really storing string data, getting the string content, converting to bytes and comparing based on bytes. This isn't quite the same thing as originally storing bytes and then comparing bytes on the way out.

For now based on our discussion this morning, we do not need such exhaustive coverage. You can limit the tests to String-only for now. I appreciate the thorough eye for tests though!